src/settings/handlers/AccountSettingsHandler.js

```diff
@@ -167,7 +168,7 @@ export default class AccountSettingsHandler extends MatrixClientBackedSettingsHa
             // This seems fishy - try and get the event for the new rooms
             const newType = this._getSettings(BREADCRUMBS_EVENT_TYPE);
             if (newType) val = newType['recent_rooms'];
-            else val = event.getContent()['rooms'];
+            else val = event.getContent()['rooms'] || [];
```

Comment:
Is this necessary, since the call to notifyUpdate already does `val || []`?  And if it is necessary, shouldn't the `val = event.getContent()['recent_rooms'];` statement below get the same treatment?

Reply:
it is indeed extra, I didn't scroll over far enough on the line below.